Paul

paulmartini opened the pull request:
This repo seems like the right place to put my code to analyze and monitor observing efficiency.

paulmartini commented:
Hi @sbailey - I know you are swamped with other stuff. Perhaps you can find another person to do the review of this PR?

sbailey reviewed (changes requested):
Thanks Paul.  I put some minor comments inline, plus a separate email comment.  More general comments:
* our desi standards is for bin/ script to _not_ have the `.py` extension, while things that are imported have the `.py` extension.
* In the context of this being installed with other desi code in the future, let's pick less generic script names than `plot_summary`, `plot_sequence`, `update_summary`, and `plot_night`, which could mean a lot of things and lead to confusion.
* similarly, several of the functions in `desisurveyops.utilities` seem specific to this use-case for survey ops, but not necessarily generically useful for other uses.  e.g. `get_twilights` is a nice generic utility function, while `calc_guidelists` seems pretty specific in what it does and what it outputs.  Consider moving some of them into a separate more focused submodule.
* Minor: Consider using `desisurveyops.utils` (like `desisurvey.utils` and `fiberassign.utils`) or `desisurveyops.util` (like `desispec.util`) rather than introducing a 3rd naming `desisurveyops.utilities`.

paulmartini commented on py/desisurveyops/utilities.py:
```diff
@@ -0,0 +1,1006 @@
+    try:
+        outdir = os.environ['DESINIGHTSTATS']
+    except KeyError:
+        outdir = '/global/cfs/cdirs/desi/users/martini/NightStats/'
```
Thanks for the general comments, Stephen. I've done the following:
- removed the .py on the bin/ scripts and gave them less generic names (plotnightobs, plotobsefficiency, plottimeobs, updateefficiency)
- moved more specific functions from utilities to obsdata.py 
- renamed utilities.py to utils.py to match the naming convention of other repos

I've also changed this specific case to report an error and raise an exception

paulmartini commented on bin/plot_summary.py:
```diff
@@ -0,0 +1,85 @@
+    hdu = fits.open(summaryfile)
+else:
+    print("Error: {} not found".format(summaryfile))
+    exit
```
Done

paulmartini commented on py/desisurveyops/utilities.py:
```diff
@@ -0,0 +1,1006 @@
+        root output directory 
+    '''
+
+    datadir = '/global/cfs/cdirs/desi/spectro/data/'
```
I've switched to `$DESI_SPECTRO_DATA`, then try `$DESI_ROOT`,  and then raise an exception if this fails. I've also renamed the function `get_rawdatadir`.

paulmartini commented on py/desisurveyops/utilities.py:
```diff
@@ -0,0 +1,1006 @@
+
+def set_outdir(verbose=False): 
+    '''
+    Set the root directory for the output and a NightlyData/ subdirectory
```
I've renamed this `get_outdir` and updated the docstring.

paulmartini commented on py/desisurveyops/utilities.py:
```diff
@@ -0,0 +1,1006 @@
+        outdir = '/global/cfs/cdirs/desi/users/martini/NightStats/'
+
+    if not os.path.isdir(outdir): 
+        os.mkdir(outdir) 
```
Done

paulmartini commented on py/desisurveyops/utilities.py:
```diff
@@ -0,0 +1,1006 @@
+    desi.date = t.strftime('%Y/%m/%d 7:00')
+
+    # Calculate astronomical twilight times
+    desi.horizon = '-18'
```
Yes, I'd been meaning to do that and appreciate the nudge/reminder. Now takes the Sun altitude as an argument.

paulmartini commented:
Hi @sbailey - Thank you for the thorough review. I think I've addressed all of your comments. Would you please take another look when you have time? Thanks!

sbailey reviewed (approved):
Thanks for the updates.  Merging now.
